diff -Naur mupdf-1.0-source-old/Makefile mupdf-1.0-source/Makefile
--- mupdf-1.0-source-old/Makefile	2012-06-11 01:05:37.000000000 -0300
+++ mupdf-1.0-source/Makefile	2012-06-11 01:05:53.000000000 -0300
@@ -12,7 +12,7 @@
 # Do not specify CFLAGS or LIBS on the make invocation line - specify
 # XCFLAGS or XLIBS instead. Make ignores any lines in the makefile that
 # set a variable that was set on the command line.
-CFLAGS += $(XCFLAGS) -Ifitz -Ipdf -Ixps -Icbz -Iscripts
+CFLAGS += $(XCFLAGS) -Ifitz -Ipdf -Ixps -Icbz -Iscripts -fpic
 LIBS += $(XLIBS) -lfreetype -ljbig2dec -ljpeg -lopenjpeg -lz -lm
 
 include Makerules
@@ -47,9 +47,9 @@
 $(OUT) $(GEN) :
 	$(MKDIR_CMD)
 
-$(OUT)/%.a :
-	$(AR_CMD)
-	$(RANLIB_CMD)
+$(OUT)/%.so.$(MUPDF_LIB_VERSION) :
+	$(LINK_CMD) -shared -Wl,-soname,$(notdir $@)
+	ln -s $(notdir $@) $(subst .$(MUPDF_LIB_VERSION),,$@)
 
 $(OUT)/% : $(OUT)/%.o
 	$(LINK_CMD)
@@ -73,7 +73,7 @@
 
 # --- Fitz, MuPDF, MuXPS and MuCBZ library ---
 
-FITZ_LIB := $(OUT)/libfitz.a
+FITZ_LIB := $(OUT)/libfitz.so.$(MUPDF_LIB_VERSION)
 
 FITZ_SRC := $(notdir $(wildcard fitz/*.c draw/*.c))
 FITZ_SRC := $(filter-out draw_simple_scale.c, $(FITZ_SRC))
diff -Naur mupdf-1.0-source-old/Makerules mupdf-1.0-source/Makerules
--- mupdf-1.0-source-old/Makerules	2012-06-11 01:05:37.000000000 -0300
+++ mupdf-1.0-source/Makerules	2012-06-11 01:05:58.000000000 -0300
@@ -4,6 +4,7 @@
 OS := $(OS:MINGW%=MINGW)
 
 CFLAGS += -Wall
+MUPDF_LIB_VERSION=1.0
 
 ifeq "$(build)" "debug"
 CFLAGS += -pipe -g -DDEBUG
